teuthology/task/internal/redhat.py

```diff
@@ -0,0 +1,143 @@
+"""
+Internal tasks  for redhat downstream builds
+"""
+import contextlib
+import logging
+import requests
+from tempfile import NamedTemporaryFile
+from teuthology.parallel import parallel
+from teuthology.orchestra import run
+from teuthology.task.install.redhat import set_deb_repo
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def setup_cdn_repo(ctx, config):
+    """
+     setup repo if set_cdn_repo exists in config
+     redhat:
+      set_cdn_repo:
+        rhbuild: 2.0 or 1.3.2 or 1.3.3
+    """
+    # do import of tasks here since the qa task path should be set here
+    if ctx.config.get('redhat').get('set-cdn-repo', None):
```

Comment:
If `redhat` is undefined, you'll still get an `AttributeError` here, so the first `get()` won't help much